### PR TITLE
Add IsDungeonBoss

### PR DESCRIPTION
**Pull request by BuildingOutLoud**
My first Git PR. Let me know if I did anything wrong. :)

### CODE REVIEW

**Comment by Rochet2** [comment]
I dont think IsWorldBoss is quite the same as IsDungeonBoss.
This is because for example mangos and TC have IsWorldBoss, but only TC has IsDungeonBoss.
https://github.com/mangoszero/server/blob/cc27f3274461663d370eaab041896bf11a6008ff/src/game/Object/Creature.h#L579
https://github.com/TrinityCore/TrinityCore/blob/f759809d9d4364bc1d988e4390d3d5a33d5469e9/src/server/game/Entities/Creature/Creature.h#L118
https://github.com/TrinityCore/TrinityCore/blob/f759809d9d4364bc1d988e4390d3d5a33d5469e9/src/server/game/Entities/Creature/Creature.h#L103

IsDungeonBoss could be added, but I think it would need to be TC/AC-specific then.
Considering all cores probably have flags_extra field in creature template, I would maybe rather have a way to get flags_extra and let people test the existence of any flags they want - even custom ones.

**Comment by BuildingOutLoud** [comment]
That's completely fair. I was told in the Mangos discord that IsWorldBoss did the same thing as IsDungeonBoss. I agree that a function to get flags_extra would be excellent. I was quite surprised when I found the comment in the AzerothCore code not to set the boss flag in the database, as it would cause problems. However, as a beginner, I don't think I'm up to the task of creating that quite yet. :)

Should I wrap the whole IsDungeonBoss statement in #if defined(TRINITY) || defined(AZEROTHCORE), or should I just close this PR out for now?

**Comment by Rochet2** [comment]
Hmm, I would maybe do the ifdef change so it's only for the specific cores. Needs to be done in both files.
I think Foe was also saying this in the discord. Maybe he meant that scripters can use IsWorldBoss in their code on mangos for the same purpose instead of invoking IsDungeonBoss at all.

> However, as a beginner, I don't think I'm up to the task of creating that quite yet. :)

Are you talking about the extra_flags get function?
It would just be like
```cpp
#if defined(TRINITY) || defined(AZEROTHCORE)
Eluna::Push(L, creature->GetCreatureTemplate()->flags_extra);
#else
Eluna::Push(L, creature->GetCreatureInfo()->ExtraFlags);
#endif
```

File: LuaFunctions.cpp [hunk 1]
```diff
@@ -809,6 +809,7 @@ ElunaRegister<Creature> CreatureMethods[] =
     { "SetEquipmentSlots", &LuaCreature::SetEquipmentSlots },
 
     // Boolean
+    { "IsDungeonBoss", &LuaCreature::IsWorldBoss },
```
Comment (Rochet2) [inline LuaFunctions.cpp:812]: This requires to be surrounded with ifdef as well. Also, looks like there is a copy-paste mistake with the `LuaCreature::IsWorldBoss`.
```suggestion
#if defined(TRINITY) || defined(AZEROTHCORE)
    { "IsDungeonBoss", &LuaCreature::IsDungeonBoss},
#endif
```

**Pull request merged**